net/dtlslistener.go

```
@@ -67,24 +85,24 @@ func NewDTLSListener(network string, addr string, cfg *dtls.Config, heartBeat ti
 // AcceptWithContext waits with context for a generic Conn.
 func (l *DTLSListener) AcceptWithContext(ctx context.Context) (net.Conn, error) {
 	for {
-		if atomic.LoadUint32(&l.closed) == 1 {
-			return nil, ErrServerClosed
-		}
 		select {
 		case <-ctx.Done():
 			return nil, ctx.Err()
 		default:
 		}
+		if atomic.LoadUint32(&l.closed) == 1 {
+			return nil, ErrListenerIsClosed
+		}
 		err := l.SetDeadline(time.Now().Add(l.heartBeat))
 		if err != nil {
-			return nil, fmt.Errorf("cannot set deadline to accept connection: %v", err)
+			return nil, fmt.Errorf("cannot set deadline to accept connection: %w", err)
 		}
 		rw, err := l.Accept()
 		if err != nil {
 			if isTemporary(err) {
 				continue
```

log a debug message when discarding an error message and handling the error

---

// check context in regular intervals and then resume listening